test/Chainweb/Test/Pact/SPV.hs

```diff
@@ -180,94 +191,44 @@ roundtrip
     -> CreatesGenerator
       -- ^ create tx generator
     -> IO (CutOutputs, CutOutputs)
-roundtrip sid0 tid0 burn create =
-  withAll v $ \sqlenv0s -> withAll v $ \sqlenv1s -> withAll v $ \sqlenv2s -> do
-    -- Pact service that is used to initialize the cut data base
-    let pactIO bhdb pdb = testWebPactExecutionService v
-                              (return bhdb) (return pdb)
-                              (return mempty) sqlenv0s
-    withTempRocksDb "chainweb-spv-tests" $ \rdb ->
-        withTestCutDb rdb v 20 pactIO logg $ \cutDb -> do
-            cdb <- newMVar cutDb
-
-            sid <- mkChainId v sid0
-            tid <- mkChainId v tid0
-
-            -- track the continuation pact id
-            pidv <- newEmptyMVar @PactId
-
-            let pdb = _webBlockPayloadStoreCas $
-                      view cutDbPayloadStore cutDb
+roundtrip sid0 tid0 burn create = do
+  withTestBlockDb v $ \bdb@(TestBlockDb wdb pdb _) -> withAll v $ \sqlenvs -> do
 
-            let webStoreDb = view cutDbWebBlockHeaderStore cutDb
-            let webHeaderDb = _webBlockHeaderStoreCas webStoreDb
-            let payloadDb = view cutDbPayloadCas cutDb
+    sid <- mkChainId v sid0
+    tid <- mkChainId v tid0
 
-            -- pact service, that is used to extend the cut data base
-            t1 <- getCurrentTimeIntegral
-            txGen1 <- burn t1 pidv sid tid
+    -- track the continuation pact id
+    pidv <- newEmptyMVar @PactId
 
-            pact1 <- testWebPactExecutionService v
-                         (return webHeaderDb) (return pdb)
-                         (chainToMPA txGen1) sqlenv1s
-            syncPact cutDb pact1
+    tg <- newMVar mempty
+    pact <- testWebPactExecutionService v
+              (return wdb) (return pdb)
+              (chainToMPA' tg) sqlenvs
 
-            c0 <- _cut cutDb
+    -- cut 0: empty run (not sure why this is needed but test fails without it)
+    void $ runCut bdb pact
```

Comment:
Not sure, I'll investigate.

Reply:
it may also be just a corner case related to starting with the genesis blocks. In production networks there probably never will be normal user txs at single digit block heights.